### PR TITLE
Resource exhaustion notifications and badges

### PR DESCRIPTION
**Pull request by saltcod**

### CODE REVIEW

File: studio/components/ui/ResourceExhaustionWarningBanner/ResourceExhaustionWarningBanner.constants.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    buttonText: undefined,
+    metric: undefined,
+  },
+  disk_io_exhaustion: {
+    bannerContent: {
+      warning: {
+        title:
+          'Your project is about to deplete its Disk IO Budget, and may become unresponsive once fully exhausted',
+        description:
+          'You will need to either optimize your performance, or upgrade your compute to a larger plan.',
```
Comment (loong) [inline studio/components/ui/ResourceExhaustionWarningBanner/ResourceExhaustionWarningBanner.constants.ts:37]:
```suggestion
          'You will need to optimize your performance or upgrade your compute. Check the usage page for more recent and detailed statistics.',
```

File: studio/components/ui/ResourceExhaustionWarningBanner/ResourceExhaustionWarningBanner.constants.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+      },
+      critical: {
+        title: 'Your project has depleted its Disk IO Budget, and may become unresponsive',
+        description:
+          'You will need to either optimize your performance, or upgrade your compute to a larger plan.',
```
Comment (loong) [inline studio/components/ui/ResourceExhaustionWarningBanner/ResourceExhaustionWarningBanner.constants.ts:42]:
```suggestion
          'You will need to optimize your performance or upgrade your compute. Check the usage page for more recent and detailed statistics.',
```

File: studio/components/ui/ResourceExhaustionWarningBanner/ResourceExhaustionWarningBanner.constants.ts [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    docsUrl: 'https://supabase.com/docs/guides/platform/database-size#disk-management',
+    buttonText: undefined,
+    metric: 'disk_space',
+  },
+  cpu_exhaustion: {
+    bannerContent: {
+      warning: {
+        title: 'Your project is currently facing high CPU usage, and its performance is affected',
+        description:
+          'You will need to either optimize your performance or upgrade your compute to a larger Project',
```
Comment (loong) [inline studio/components/ui/ResourceExhaustionWarningBanner/ResourceExhaustionWarningBanner.constants.ts:92]:
```suggestion
          'You will need to optimize your performance or upgrade your compute. Check the usage page for more recent and detailed statistics.',
```

File: studio/components/ui/ResourceExhaustionWarningBanner/ResourceExhaustionWarningBanner.constants.ts [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+      },
+      critical: {
+        title: "Your project's CPU usage is at 100% and its performance is affected",
+        description:
+          'You will need to either optimize your performance or upgrade your compute to a larger plan',
```
Comment (loong) [inline studio/components/ui/ResourceExhaustionWarningBanner/ResourceExhaustionWarningBanner.constants.ts:97]:
```suggestion
          'You will need to optimize your performance or upgrade your compute. Check the usage page for more recent and detailed statistics.',
```

File: studio/components/ui/ResourceExhaustionWarningBanner/ResourceExhaustionWarningBanner.constants.ts [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    buttonText: 'Check usage',
+    metric: 'cpu',
+  },
+  memory_and_swap_exhaustion: {
+    bannerContent: {
+      warning: {
+        title:
+          'Your project is currently facing high memory usage, and its performance is affected',
+        description:
+          'You will need to either optimize your performance or upgrade your compute to a larger plan',
```
Comment (loong) [inline studio/components/ui/ResourceExhaustionWarningBanner/ResourceExhaustionWarningBanner.constants.ts:120]:
```suggestion
          'You will need to optimize your performance or upgrade your compute. Check the usage page for more recent and detailed statistics.',
```

File: studio/components/ui/ResourceExhaustionWarningBanner/ResourceExhaustionWarningBanner.constants.ts [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+      },
+      critical: {
+        title: "Your project's memory usage is at 100%, and its performance is affected",
+        description:
+          'You will need to either optimize your performance or upgrade your compute to a larger plan',
```
Comment (loong) [inline studio/components/ui/ResourceExhaustionWarningBanner/ResourceExhaustionWarningBanner.constants.ts:125]:
```suggestion
          'You will need to optimize your performance or upgrade your compute. Check the usage page for more recent and detailed statistics.',
```

**Review by loong — APPROVED** [review]
LGTM!

Just some comments on the copy's to address @soedirgo concerns.

File: studio/components/interfaces/Home/ProjectList/ProjectCard.tsx [hunk 7]
```diff
@@ -1,55 +1,71 @@
...
 
 export interface ProjectCardProps {
   project: Project
   rewriteHref?: string
   githubIntegration?: IntegrationProjectConnection
   vercelIntegration?: IntegrationProjectConnection
+  resourceWarnings?: ResourceWarning
+}
+
+const getProjetStatus = (project: Project) => {
```
Comment (joshenlim) [inline studio/components/interfaces/Home/ProjectList/ProjectCard.tsx:18]: nit: spelling

File: studio/components/interfaces/Home/ProjectList/ProjectCardStatus.tsx [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+    (x) => resourceWarnings[x as keyof typeof resourceWarnings] === 'critical'
+  )
+
+  const isCritical = activeWarnings.includes('is_readonly_mode_enabled') || hasCriticalWarning
+
+  const getWarningType = () => {
+    let type = 'default'
+
+    switch (true) {
+      case projectStatus === 'isPaused':
```
Comment (joshenlim) [inline studio/components/interfaces/Home/ProjectList/ProjectCardStatus.tsx:58]: is this correct? `isPaused` or `isPausing`?

**Pull request merged**